modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -222,8 +222,14 @@ private static Class<?> loadClass(final String className, final Object instance)
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
      * {@code Control} via the {@link Skin#getSkinnable()} method.
      * <p>
+     * To ensure a one-to-one relationship between a {@code Control} and its {@code Skin},
+     * {@link Control#setSkin(Skin)} will check the return value of {@link Skin#getSkinnable()}
```

Comment:
strictly speaking, it's not the method setSkin but the property skin ..

Reply:
please check the updated comment, I think it sounds weird...

Reply:
that change sounds okay to me :) 

What might be missing is a description of that it actually does, that is taking full control about the old.dispose / new.install sequence as spec'ed in Skin - I think it could be repeated here to emphasize that it's living up to its responsibility.

Reply:
I think it's a good idea, and a good spot for the explanation.